user_guide_src/source/general/configuration.rst

```diff
@@ -15,6 +15,24 @@ the application configuration files in the **app/Config** folder.
     :local:
     :depth: 2
 
+
+What are Configuration Classes?
+*******************************
+
+Configuration classes are utilized to define system default configuration values.
+System configuration values are typically *static*. Configuration classes are
+intended to retain the settings that configure how the application operates,
+rather than responding to each user's individual settings.
+
+It is not recommended to alter values set during the instantiation of a
+configuration class later during execution. In other words, it is recommended to
+treat configuration classes as immutable or readonly classes. This is especially
+important if you utilize :ref:`factories-config-caching`.
+
+Configuration values can be retrieved from environment variables, hard-coded
+within the class files, or stored in a database using the :ref:`Settings <settings>`
+library.
```

Comment:
This explanation is misleading because the Setting library does not affect the instantiation of Config objects.
I will reconsider.